pangeo_forge_recipes/transforms.py

```diff
@@ -487,11 +521,16 @@ def expand(
             target=self.get_full_target(), target_chunks=self.target_chunks
         )
         n_target_stores = rechunked_datasets | StoreDatasetFragments(target_store=target_store)
+
         singleton_target_store = (
-            n_target_stores
-            | beam.combiners.Sample.FixedSizeGlobally(1)
-            | beam.FlatMap(lambda x: x)  # https://stackoverflow.com/a/47146582
+            n_target_stores | SampleSingleton()
+            if not self.consolidate_coords
+            else n_target_stores | SampleSingleton() | ConsolidateDimensionCoordinates()
         )
-        # TODO: optionally use `singleton_target_store` to
-        # consolidate metadata and/or coordinate dims here
+        singleton_target_store = (
+            n_target_stores | SampleSingleton()
+            if not self.consolidate_metadata
+            else n_target_stores | SampleSingleton() | ConsolidateZarrMetadata()
+        )
+
         return singleton_target_store
```

Comment:
```suggestion
        return (
            singleton_target_store
            if not self.consolidate_metadata
            else singleton_target_store | ConsolidateZarrMetadata()
        )
```

Reply:
> The last bit of logic in StoreToZarr could probably be combined. @cisaacstern any thoughts here?

@norlandrhagen, awesome progress! Haven't done a full review but above is my suggestion regarding this point.